### PR TITLE
CON2508: Cleanup of stale multipath devices for node monitor as well as node init container

### PR DESCRIPTION
**Pull request by AnushaY1916**
The following functionalities are implemented in phase 3:

1. Unmount all references to the device, bind mounts, and real mounts.
2. Kill processes using the mount points and devices
3. Remove block devices
4. Flush the multipath map
5. Enable and disable of node monitor and cleanup in the node init container based on the user's choice

made improvements for the previous  phase's implementation as well

### CODE REVIEW

**Review by dileepds — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -22,22 +22,22 @@ FROM registry.access.redhat.com/ubi9/ubi-minimal:9.4-949.1714662671
...
-      io.k8s.description="The HPE CSI Driver for Kubernetes enables container orchestrators, such as Kubernetes and OpenShift, to manage the life-cycle of persistent storage." \
-      io.openshift.tags=hpe,csi,hpe-csi-driver
+    maintainer="HPE Storage" \
+    vendor="HPE" \
+    version="2.5.0-beta" \
+    summary="HPE CSI Driver for Kubernetes" \
+    description="The HPE CSI Driver for Kubernetes enables container orchestrators, such as Kubernetes and OpenShift, to manage the life-cycle of persistent storage." \
+    io.k8s.display-name="HPE CSI Driver for Kubernetes" \
+    io.k8s.description="The HPE CSI Driver for Kubernetes enables container orchestrators, such as Kubernetes and OpenShift, to manage the life-cycle of persistent storage." \
+    io.openshift.tags=hpe,csi,hpe-csi-driver
```
Comment (datamattsson) [inline Dockerfile:40]: Why change the indent here?

File: Dockerfile [hunk 2]
```diff
@@ -22,22 +22,22 @@ FROM registry.access.redhat.com/ubi9/ubi-minimal:9.4-949.1714662671
 RUN microdnf update -y && rm -rf /var/cache/yum
 ADD cmd/csi-driver/AlmaLinux-Base.repo /etc/yum.repos.d/
 
-RUN microdnf install -y cryptsetup tar procps
+RUN microdnf install -y cryptsetup tar procps psmisc
```
Comment (datamattsson) [inline Dockerfile:25]: Shouldn't we borrow `fuser` from the host? That's what I did.

**Review by datamattsson — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (AnushaY1916) [reply Dockerfile:40]: The IDE updated it when saved.

File: Dockerfile (hunk 2 above)
Comment (AnushaY1916) [reply Dockerfile]: Updated

**Review by datamattsson — APPROVED** [review]

**Pull request merged**